sw/airborne/modules/dragspeed/dragspeed.c

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright (C) Tom van Dijk
+ *
+ * This file is part of paparazzi
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+/**
+ * @file "modules/dragspeed/dragspeed.c"
+ * @author Tom van Dijk
+ * This module estimates the velocity of rotorcraft by measuring the drag force using the accelerometer.
+ */
+
+#include "modules/dragspeed/dragspeed.h"
+
+#include "subsystems/abi.h"
+#include "subsystems/abi_common.h"
+#include "subsystems/datalink/telemetry.h"
+
+#ifndef DRAGSPEED_ACCEL_ID
+#define DRAGSPEED_ACCEL_ID ABI_BROADCAST
+#endif
+
+#ifndef DRAGSPEED_COEFF_X
+#define DRAGSPEED_COEFF_X 1.0 /// Drag coefficient (mu/m) of the linear drag model along x axis, where m*a = v*mu
+#endif
+
+#ifndef DRAGSPEED_COEFF_Y
+#define DRAGSPEED_COEFF_Y 1.0 /// Drag coefficient (mu/m) of the linear drag model along y axis, where m*a = v*mu
+#endif
+
+#ifndef DRAGSPEED_ZERO_X
+#define DRAGSPEED_ZERO_X 0.0 /// Accelerometer reading (x axis) when stationary [m/s^2]
+#endif
+
+#ifndef DRAGSPEED_ZERO_Y
+#define DRAGSPEED_ZERO_Y 0.0 /// Accelerometer reading (y axis) when stationary [m/s^2]
+#endif
+
+#ifndef DRAGSPEED_R
+#define DRAGSPEED_R 0.25 /// Velocity measurement noise variance [(m/s)^2]
+#endif
+
+#ifndef DRAGSPEED_FILTER
+#define DRAGSPEED_FILTER 0.8 /// First-order low-pass filter strength [0..1]. Pretty high default value as accelero's tend to be noisy.
+#endif
+
+struct dragspeed_t dragspeed;
+
+static abi_event accel_ev;
+static void accel_cb(
+		uint8_t sender_id,
+		uint32_t stamp,
+		struct Int32Vect3 *accel);
+
+static void send_dragspeed(struct transport_tx *trans, struct link_device *dev);
+
+static void calibrate_coeff(struct Int32Vect3 *accel);
+static void calibrate_zero(struct Int32Vect3 *accel);
+
+void dragspeed_init(void) {
+	// Set initial values
+	dragspeed.coeff.x = DRAGSPEED_COEFF_X;
+	dragspeed.coeff.y = DRAGSPEED_COEFF_Y;
+	dragspeed.filter = DRAGSPEED_FILTER;
+	// Register callbacks
+	register_periodic_telemetry(DefaultPeriodic, PPRZ_MSG_ID_DRAGSPEED,
+			send_dragspeed);
+	AbiBindMsgIMU_ACCEL_INT32(DRAGSPEED_ACCEL_ID, &accel_ev, accel_cb);
+}
+
+bool dragspeed_calibrate_coeff(void) {
+	dragspeed.do_calibrate_coeff = 1;
+	return FALSE;
```

Comment:
Run fix_code_style. You can also add the pprz code style to your IDE so this doesn't happen in the future. There are project guidelines for eclipse, sublime and qt in the head folder (or the tools sub-folder). If you use a different IDE you can refer to the code style here http://docs.paparazziuav.org/latest/stylec.html